### PR TITLE
feat: update schema component

### LINKED ISSUE #6: Schema
Get the structure of the documents in a collection
- Parse document types as tables
- Parsing the types to determine the nullability of the fields
- Capitalize the names of the documents.

### PR DESCRIPTION
**Pull request by yoanesp**
- parse schema from bucket, scope and collection
 - update config
 - update query 
  
 close #6

### CODE REVIEW

**Review by biozal — CHANGES_REQUESTED** [review]
More feedback:

In general, method signatures should match the documentation.  Examples of breaking this rule include:
- In query.ts lines, 353 - 354 parameters don't match the function signature.  
- In query.ts some functions have documentation of parameters and some don't.  We should document every function.  

In general, do not leave code comments out in the codebase.  Here are some examples:

- In query.ts lines 53 - 61 are commented out with no idea why they are still in the code.  
- In query.ts lines 90 - 95 are commented out in the switch statement and you are just throwing back an error.
- In query.ts lines 224 - 260 are commented out and no idea why.
- In query.ts lines 282 - 283 are commented out.
- In query.ts lines 359 - 366 are commented out.
- In query.ts lines 445 - 490 are commented out and not sure if this is documentation or not because trying to understand how it relates to the code below it.  

This is co-reviewed by @ejscribner

**Comment by yoanesp** [comment]
> More feedback:
> 
> In general, method signatures should match the documentation. Examples of breaking this rule include:
> 
> * In query.ts lines, 353 - 354 parameters don't match the function signature.
> * In query.ts some functions have documentation of parameters and some don't.  We should document every function.
> 
> In general, do not leave code comments out in the codebase. Here are some examples:
> 
> * In query.ts lines 53 - 61 are commented out with no idea why they are still in the code.
> * In query.ts lines 90 - 95 are commented out in the switch statement and you are just throwing back an error.
> * In query.ts lines 224 - 260 are commented out and no idea why.
> * In query.ts lines 282 - 283 are commented out.
> * In query.ts lines 359 - 366 are commented out.
> * In query.ts lines 445 - 490 are commented out and not sure if this is documentation or not because trying to understand how it relates to the code below it.
> 
> This is co-reviewed by @ejscribner

File: src/schema.ts [hunk 1]
```diff
@@ -1,41 +1,54 @@
-import { SchemaResponse, TableInfo } from "@hasura/dc-api-types";
+import { ColumnInfo, SchemaResponse, TableInfo } from "@hasura/dc-api-types";
 import { Cluster } from "couchbase";
 import { Config } from "./config";
 
-/*const DocumentInfoSchema = new Schema({
```
[the remark this answers is not in the document]
Comment (yoanesp) [reply src/schema.ts:5]: 👍

File: src/query.ts [hunk 2]
```diff
@@ -354,7 +355,7 @@ import { coerceUndefinedOrNullToEmptyRecord, coerceUndefinedToNull, envToBool, e
   * @returns string representing the combined where clause
   */
  function where(whereExpression: Expression | null, queryTableName: TableName, queryTableAlias: string): string {
-   const whereClause = whereExpression !== null ? [where_clause(whereExpression, queryTableName, queryTableAlias)] : [];
+   const whereClause = whereExpression !== null ? [ `type = "${queryTableAlias}"` , where_clause(whereExpression, queryTableName, queryTableAlias)] : [ `type = "${queryTableAlias}"`];
    /*const joinArray = joinInfo
```
[the remark this answers is not in the document]
Comment (yoanesp) [reply src/query.ts]: No, I don't

File: src/config.ts [hunk 3]
```diff
@@ -3,8 +3,8 @@ import { ConfigSchemaResponse } from "@hasura/dc-api-types"
 
 export type Config = {
   bucket: string,
-  scope: string,
-  collections: String[] | null
+  scope: string | '_defualt',
```
[the remark this answers is not in the document]
Comment (yoanesp) [reply src/config.ts]: I updated this

**Review by biozal — APPROVED** [review]

**Review by ejscribner — APPROVED** [review]

**Pull request merged**